lib/transform.js

```diff
@@ -136,8 +114,7 @@ exports = module.exports = function transform(content, mode, maxHeaderLevel, tit
   var currentToc = info.hasStart && lines.slice(info.startIdx, info.endIdx).join('\n')
     , linesToToc = getLinesToToc(lines, currentToc, info);
 
-  var headers = getHashedHeaders(linesToToc, maxHeaderLevel)
-    .concat(getUnderlinedHeaders(linesToToc, maxHeaderLevel))
+  var headers = getMarkdownHeaders(linesToToc, maxHeaderLevel)
     .concat(getHtmlHeaders(linesToToc, maxHeaderLevelHtml))
```

Comment:
Do we still need to parse HTML headers ourselves? I would have hoped that `markdown-to-ast` would parse that for us, but maybe not.

If it doesn't, do we want to do anything special to handle the case of `<h1><a href="#">Header</a><h1>`? I'm fine with not supporting this edge case.